source/extensions/tracers/xray/tracer.cc

```diff
@@ -155,19 +162,8 @@ Tracing::SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::str
   return span_ptr;
 }
 
-XRay::SpanPtr Tracer::createNonSampledSpan(const Tracing::Config& config,
-                                           const std::string& operation_name,
-                                           Envoy::SystemTime start_time,
-                                           const absl::optional<XRayHeader>& xray_header) const {
+XRay::SpanPtr Tracer::createNonSampledSpan(const absl::optional<XRayHeader>& xray_header) const {
```

Comment:
As part of earlier PR [#20400](https://github.com/envoyproxy/envoy/pull/20400) I had added all these args into method `createNonSampledSpan`. But now I realized that we only need xray_header for `Span::injectContext` [L104](https://github.com/envoyproxy/envoy/blob/main/source/extensions/tracers/xray/tracer.cc#L104) for future, but nothing else since inside method `Span::finishSpan` [L58](https://github.com/envoyproxy/envoy/blob/main/source/extensions/tracers/xray/tracer.cc#L58) we don't use any of these additional values if we are not sampling the segment.